ipatests/test_integration/test_sudo.py

```diff
@@ -153,8 +153,15 @@ def test_advise_script_enable_sudo_admins(self):
                 'rules cleanup failed'
 
     def test_nisdomainname(self):
-        result = self.client.run_command('nisdomainname')
-        assert self.client.domain.name in result.stdout_text
+        """
+            Test that the client has been set up with proper NIS domain name
+            Also support containers where NIS domain name cannot be used
+            in runc environments
+        """
+        result = self.client.run_command('nisdomainname', raiseonerr=False)
+        assert (self.client.domain.name in result.stdout_text or
+                (("Local domain name not set" in result.stdout_text) and
+                 result.returncode == 1))
```

Comment:
I'd prefer if we detect that this is a container and only allow "Local domain not set" in that case otherwise this could regress and we wouldn't know.

Reply:
Correct, I'll add another check. Or maybe move to skip the test in container.